packages/core/src/shared/utilities/pathUtils.ts

```diff
@@ -126,3 +128,24 @@ export function getDriveLetter(path: string): string {
 
     return fullpath.substring(0, 1)
 }
+
+/**
+ * Checks if the provided path is valid and exists.
+ * @todo: Migrate fs.existsSync to fsCommon.exists
+ *
+ * @param {string} [path] - The path to be checked. If not provided, defaults to an empty string.
+ * @returns {boolean} Returns true if the path is valid and exists, otherwise returns false.
+ */
+export function isValidPath(path?: string): boolean {
```

Comment:
this would never be hit in web-mode because `getHomeDirectory` returns early for web-mode

Reply:
`fs` and `fsCommon` is not supported in `web` where as `fs-extra` only has a promise based API.

Thus, if we don't return early for web-mode, we will have update with a `Promise` based API.

Reply:
I pushed a change to introduce `initUserHomeDir()`, which is a one-time routine done at startup that validates $HOME and other env vars. That allows `initUserHomeDir()` to be async while `getUserHomeDir()` remains "sync". As a minor perf bonus, `getUserHomeDir()` now uses a cached value.